src/main/native/smithwaterman/PairWiseSW.h

```diff
@@ -410,20 +409,35 @@ void inline getCIGAR(SeqPair *p, int16_t *cigarBuf_, int32_t tid)
                 state = 'R';
                 break;
         }
-	// expectedLength for converting int to str w/ extra padding for '\0'
-	int expectedLength = snprintf( NULL, 0, "%d%c", cigarArray[2 * i + 1], state);
-	if (curSize >= 0 && expectedLength > 1 && (curSize < maxSize - expectedLength ))
-           {
-                   curSize += snprintf(p->cigar + curSize, expectedLength + 1, "%d%c", cigarArray[2 * i + 1], state);
-           }
+        // expectedLength for converting int to str w/ extra padding for '\0'
+	    int expectedLength = snprintf( NULL, 0, "%d%c", cigarArray[2 * i + 1], state);
+        if (curSize >= 0 && expectedLength > 1){ 
+            
+            int tempNewCurSize = curSize + expectedLength;
+
+            if (tempNewCurSize < cigarBufLength)
+            {
+                curSize += snprintf(p->cigar + curSize, expectedLength + 1, "%d%c", cigarArray[2 * i + 1], state);
+            }
+
+            else if (tempNewCurSize == cigarBufLength)
+            {
+                char * tempBuff = (char*) malloc(sizeof(char) * (expectedLength + 1));
+                snprintf(tempBuff, expectedLength + 1, "%d%c", cigarArray[2 * i + 1], state);
+                
+                strncpy(p->cigar + curSize, tempBuff, expectedLength);
+                curSize += expectedLength;
```

Comment:
As the cigarBuf size is the upper limit, does the last byte \0 gets truncated here?

Reply:
Yes, It gets truncated. The returned cigar is the type of byte[] so there is no need to have a trailing 0.